geode-core/src/main/java/org/apache/geode/internal/cache/HttpService.java

```diff
@@ -64,7 +64,7 @@
 
   public HttpService(String bindAddress, int port, SSLConfig sslConfig) {
     if (port == 0) {
-      return;
+      throw new IllegalArgumentException("HttpService not started with port = 0.");
```

Comment:
I don't like this change. Our docs say that setting http-service-port to 0 disables the HTTP service.
But throwing this exception will cause code in GemFireCacheImpl to log a warning. Setting the property to 0 should not cause a warning.
I suggest that instead of throwing an exception here you log an info message about the service being disabled and just return as the code did before.
The other option would be to change this code in GemFireCacheImpl to not even call new HttpService if systemConfig.getHttpServicePort() is zero. Maybe that is better because it seems odd for GemFireCacheImpl to have an instance of HttpService if it has been disabled.

Reply:
Here is the code in GemFireCacheImpl:        try {
          httpService = Optional.of(new HttpService(systemConfig.getHttpServiceBindAddress(),
              systemConfig.getHttpServicePort(), SSLConfigurationFactory
                  .getSSLConfigForComponent(systemConfig, SecurableCommunicationChannel.WEB)));
        } catch (Throwable ex) {
          logger.warn("Could not enable HttpService: {}", ex.getMessage());
        }

Reply:
Noted. I just want to httpService to be null if port=0.